src/OpenTelemetry.Extensions/Trace/BaggageActivityProcessor.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace;
+
+/// <summary>
+/// Activity processor that adds <see cref="Baggage"/> fields to every new span.
+/// </summary>
+internal sealed class BaggageActivityProcessor : BaseProcessor<Activity>
+{
+    /// <inheritdoc />
+    public override void OnStart(Activity activity)
```

Comment:
@MikeGoldsmith Should we check `Activity.IsAllDataRequested` before adding the tags?

Reply:
Not needed right? processors won't be called at all without that